src/libAtomVM/externalterm.c

```diff
@@ -221,7 +223,56 @@ static void write_bytes(uint8_t *buf, avm_uint64_t val)
     }
 }
 
-static int serialize_term(uint8_t *buf, term t, GlobalContext *glb)
+static bool has_extended_utf8_encoding(const uint8_t *atom_data, size_t atom_len)
+{
+    for (size_t i = 0; i < atom_len; ) {
+        size_t out_len = 0;
+        uint32_t c;
+        enum UnicodeTransformDecodeResult res = bitstring_utf8_decode(
```

Comment:
atoms in our atom table always have a valid UTF-8 encoding, so this code will never work.
If you want to know if an atom can be encoded using latin1, you need to check if each codepoint is <= 255.

Reply:
Sorry, the _purpose_ of this function is to determine whether the atom has any extended UTF8 characters, which is part of the enforcement of the following constraint in `term_to_binary/2`, when `{minor_version, 1}` is set.

From the OTP docs:

> Atoms that can be represented by a latin1 string are encoded using latin1 while only atoms that cannot be represented by latin1 are encoded using utf8.

So the idea is that if `{minor_version, 1}` is set, and if the atom cannot be encoded as latin1, it will be encoded utf8.